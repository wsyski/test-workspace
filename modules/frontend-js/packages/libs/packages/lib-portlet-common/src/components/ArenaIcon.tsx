import classNames from 'classnames';
import React from 'react';
import warning from 'warning';

export const ARENA_ICON_SPRITEMAP = '/o/images-provider/images/icons/icons.svg';
export const ArenaIconSpriteContext = React.createContext('');

interface Props extends React.SVGAttributes<SVGSVGElement> {
	className?: string;

	/**
	 * Path to the location of the spritemap resource.
	 */
	spritemap?: string;

	/**
	 * The id of the icon in the spritemap.
	 */
	symbol: string;
}

const ArenaIcon = React.forwardRef<SVGSVGElement, Props>(
	({className, spritemap, symbol, ...otherProps}: Props, ref) => {
		let spriteMapVal = React.useContext(ArenaIconSpriteContext);

		if (spritemap) {
			spriteMapVal = spritemap;
		}

		warning(
			spriteMapVal,
			'ArenaIcon requires a `spritemap` via prop or ArenaIconSpriteContext'
		);

		return (
			<svg
				{...otherProps}
				className={classNames(
					`lexicon-icon lexicon-icon-${symbol}`,
					className
				)}
				key={symbol}
				ref={ref}
				role="presentation"
			>
				<use xlinkHref={`${spriteMapVal}#${symbol}`} />
			</svg>
		);
	}
);

ArenaIcon.displayName = 'ArenaIcon';

export default ArenaIcon;
