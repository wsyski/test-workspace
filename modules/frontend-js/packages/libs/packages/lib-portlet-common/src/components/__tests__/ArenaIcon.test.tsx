import {render} from '@testing-library/react'

import React from 'react';

import '@testing-library/jest-dom'

import ArenaIcon, {ArenaIconSpriteContext} from '../ArenaIcon';

describe('ArenaIcon', () => {
    it('renders', () => {
        const {container} = render(
            <ArenaIcon
                spritemap="/path/to/some/resource.svg"
                symbol="cool-icon"
            />
        );

        expect(container).toMatchSnapshot();
    });

    it('renders with context spritemap', () => {
        const {container} = render(
            <ArenaIconSpriteContext.Provider value="foo/bar.svg">
                <ArenaIcon symbol="cool-icon"/>
            </ArenaIconSpriteContext.Provider>
        );

        expect(container).toMatchSnapshot();
    });
});
