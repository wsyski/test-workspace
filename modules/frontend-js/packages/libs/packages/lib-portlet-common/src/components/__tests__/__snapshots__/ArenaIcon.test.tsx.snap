// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArenaIcon renders 1`] = `
<div>
  <svg
    class="lexicon-icon lexicon-icon-cool-icon"
    role="presentation"
  >
    <use
      xlink:href="/path/to/some/resource.svg#cool-icon"
    />
  </svg>
</div>
`;

exports[`ArenaIcon renders with context spritemap 1`] = `
<div>
  <svg
    class="lexicon-icon lexicon-icon-cool-icon"
    role="presentation"
  >
    <use
      xlink:href="foo/bar.svg#cool-icon"
    />
  </svg>
</div>
`;
